app/javascript/src/components/Dashboard/Notes/MenuBar.jsx

```diff
@@ -0,0 +1,85 @@
+import React, { useState } from "react";
+
+import { Search, Settings, Plus } from "neetoicons";
+import { Typography } from "neetoui";
+import { MenuBar } from "neetoui/layouts";
+
+import {
+  NOTES_MENUBAR_SEGMENTS,
+  NOTES_MENUBAR_TAGS,
+  NOTES_MENUBAR_OPTIONS,
+} from "./constants";
+
+const Menu = ({ showMenu }) => {
+  const [isSearchCollapsed, setIsSearchCollapsed] = useState(true);
+
+  return (
+    <div className="flex">
+      <MenuBar showMenu={showMenu} title="Notes">
+        {NOTES_MENUBAR_OPTIONS.map(option => (
+          <MenuBar.Block
+            active={option.active}
+            count={option.count}
+            key={option.title}
+            label={option.title}
+          />
+        ))}
+        <MenuBar.SubTitle
+          iconProps={[
+            {
+              icon: Search,
+              onClick: () => setIsSearchCollapsed(!isSearchCollapsed),
```

Comment:
https://academy.bigbinary.com/learn-react/pages/know-when-to-use-previous-state-for-state-update